llvm/include/llvm/IR/DebugInfo.h

```diff
@@ -40,7 +40,7 @@ class Module;
 
 /// Finds dbg.declare intrinsics declaring local variables as living in the
 /// memory that 'V' points to.
-TinyPtrVector<DbgDeclareInst *> FindDbgDeclareUses(Value *V);
+void findDbgDeclares(SmallVectorImpl<DbgDeclareInst *> &DbgUsers, Value *V);
```

Comment:
Out parameters also encourage non-constness of variables

Reply:
I agree that avoiding out-parameters would be ideal. 

I chose out-params here for consistency. findDbgValues has always had an out parameter and recent DPValue patches added a second one for DPValues. So it seemed better to follow suit with findDbgDeclares rather than have two different interfaces for similar functions.

> this could be accomplished with multiple functions

IMO that would make it easier for the old and new debug modes to diverge - having a single "find all debug-things" function seemed to make sense from that point of view.

One final point - we're in a bit of a transitionary stage. We would like to be able to completely remove debug intrinsics soon, at which point these functions go back to having a single "return value". Perhaps worth refactoring at that point?

I don't have a really strong opinion on this - I just wanted to show my working so far.

Reply:
FYI

>Forgive me if I mentioned this in another patch (I couldn't find it), but why insist on using out parameters instead of returning a pair of vectors?

It was in #74480, for a similar function.

Reply:
Since this is a transitionary state, this is probably fine. But you know how LLVM's transitionary states can be usually last a few years...

> I chose out-params here for consistency. findDbgValues has always had an out parameter

Consistency is a good argument for sure. It is a bit unfortunate because we had the chance to modify the bad API and make it identical to the good API, instead of changing the good API to match the bad API. Now we have two extra steps before we can have good APIs all around

Reply:
> FYI
> 
> > Forgive me if I mentioned this in another patch (I couldn't find it), but why insist on using out parameters instead of returning a pair of vectors?
> 
> It was in #74480, for a similar function.

Ohh thank you! That reminds me I need to get back to that review. I keep clicking the "pull requests" view of Github hoping it will show all reviews I am involved in, but sadly it doesn't :(